media/js/firefox/new/variation-scene1.js

```diff
@@ -8,12 +8,35 @@ For 'standard' /new variations, this file will append all query params
 so the correct template is rendered on /download/thanks/.
 */
 
-(function($) {
+(function() {
     'use strict';
 
-    $('.download-link').each(function(i, link) {
-        if (link.href.indexOf('download/thanks/') > -1) {
-            link.href = link.href + location.search;
+    var downloadLinks = document.getElementsByClassName('download-link');
+    var href;
+    var newQs;
+    var params = new window._SearchParams().params;
+    var prefix;
+
+    // we need to propogate 'v' or 'xv' query params to ensure the correct template is loaded on /download/thanks/
+    // only one of these keys should be present
+
+    // 'xv' denotes a 'variant' page for a specific marketing campaign
+    if (params.hasOwnProperty('xv')) {
+        newQs = 'xv=' + params['xv'];
+    // 'v' denotes an experiment variation
+    } else if (params.hasOwnProperty('v')) {
```

Comment:
Should this be an `if` instead of an `else if`? Currently if there's both an `xv` param and `v` param, only the former gets appended and the latter gets dropped.

Reply:
I originally had this as two separate `if` statements, but then thought that there's no way both `v` and `xv` can meaningfully co-exist.

Buuuuut, maybe if an `xv` has an experiment running with different `v` values, they could? I'll switch back to two separate `if` statements. Worst case scenario is one clobbers the other - more of a task for the view in that case.

Reply:
Experience variations can (and sometimes have) had multiple A/B test variations, so yes I think this is something we'll need to support 👍